crates/ty_python_semantic/src/semantic_index/predicate.rs

```diff
@@ -150,6 +150,9 @@ pub(crate) struct PatternPredicate<'db> {
     pub(crate) kind: PatternPredicateKind<'db>,
 
     pub(crate) guard: Option<Expression<'db>>,
+
+    /// A reference to the pattern of the previous match case
+    pub(crate) previous_predicate: Option<Box<PatternPredicate<'db>>>,
```

Comment:
So there are probably more efficient ways to implement this, but this linked-list approach was so unintrusive..